Add clear button to send screen and fix Input for small text

MuckT opened the pull request:
### Description

Includes a button for clear functionality, tapping back numerous times was tedious.
Included styling to handle large numbers. On iOS values is shown with ellipses while retaining the value. On Android the value is shown in entirely, but in a scrollable view. The value is not impacted by these style changes.

### Other changes

N/A

### Tested

Tested locally on iOS and Android.

### How others should test

- Using the Send Amount component tap max and ensure the value displays without broken numbers or super small text.
- Copy the value and ensure that the whole value is sent.
- Tap 'CLEAR' and ensure the value is cleared. 

### Related issues

N/A

### Backwards compatibility

This change is backwards compatible.

### Screenshot
<div>
  <div>
    <h2>Android Send Amount</h2>
    <img src="https://user-images.githubusercontent.com/26950305/142981740-79db1d57-4bef-4f33-b613-c0b9c395ee8e.png" alt="Android-SendAmount" width="45%" height="auto">
  </div>
  <div>
    <h2>iOS Send Amount</h2>
    <img src="https://user-images.githubusercontent.com/26950305/142982064-3a4102dc-8f51-416f-9e69-16d175d293c9.png" alt="iOS-SendAmount" width="45%" height="auto">
  </div>
</div>

### Notes

- This change doesn't match the mocks, but after using the send amount screen a clear input and larger text was needed.

kathaypacific commented on packages/mobile/src/send/SendAmount/SendAmountValue.tsx:
```diff
@@ -41,9 +45,16 @@ function SendAmountValue({
-        <BorderlessButton style={styles.maxButtonContainer} onPress={onPressMax}>
-          <Text style={styles.maxButton}>{t('max')}</Text>
-        </BorderlessButton>
+        {isOutgoingPaymentRequest ? null : (
```
just checking that we want to render null instead of the "max" button if this is true, as the "max" button always existed in the previous version?

MuckT replied:
The max button is new in 1.22.0. We don't want to render the max button on outgoing payment request.

kathaypacific commented on packages/mobile/src/send/SendAmount/SendAmountValue.tsx:
```diff
@@ -54,7 +65,15 @@ function SendAmountValue({
+                numberOfLines={1}
+                minimumFontScale={0.4}
+                selectable={true}
+                ellipsizeMode={'tail'}
```
sorry this is really nitpicky but can this be `ellipsizeMode="tail"`?

kathaypacific commented on packages/mobile/src/send/SendAmount/SendAmountValue.tsx:
```diff
@@ -102,6 +121,10 @@ const styles = StyleSheet.create({
     alignItems: 'center',
   },
+  optionsContainer: {
+    flexDirection: 'column',
```
i think flexDirection is column by default so maybe not necessary here? :)

kathaypacific reviewed (commented):
I think this looks good, and the clear button seems useful :) do we need to check in with Aleksey/Nitya before this can be merged?

MuckT commented:
@kathaypacific looking over the review notes now. Having sign-off from Aleksey/Nitya would be needed before this is merged.

kathaypacific commented on packages/mobile/src/send/SendAmount/SendAmountValue.tsx:
```diff
@@ -116,29 +128,27 @@ const styles = StyleSheet.create({
     ...fontStyles.regular,
     fontSize: 64,
     lineHeight: undefined,
+    fontFamily: 'arial',
```
just checking, is it possible to use the Inter or Jost fonts from `fonts.tsx`? if i understand the workaround for the above github issue correctly, we can still use our app fonts?

kathaypacific reviewed (approved):
The code changes look fine to me :) I do see that there are quite a few places where you've changed the number of decimal places to show for certain token/currency values - I assume you've checked the business need of these with Aleksey (I don't have enough business knowledge to have thoughts about this, sorry!)

kathaypacific commented:
> @kathaypacific looking over the review notes now. Having sign-off from Aleksey/Nitya would be needed before this is merged. Removed the clear button and move the functionality to a long press on the backspace button. **Edit**: Checked with Aleksey and I ended up moving the clear functionality to a longPress on the backspace button.

Sorry for the late response @MuckT I didn't get a notification about your comment edit - I thought you were still checking with Aleksey about this! Just noticed it this morning.

MuckT commented:
@kathaypacific I've switched over the font family to 'Jost-Medium', switched max to be 8 digits, and improved some handling of the screen with large text sizes.

kathaypacific commented:
> @kathaypacific I've switched over the font family to 'Jost-Medium', switched max to be 8 digits, and improved some handling of the screen with large text sizes.

Nice! Ship it 🚢

ValoraQA commented:
Hey @MuckT, We have verified this tesk using Latest iOS Alfajores Nightly V1.23.0, Android Alfa jores nightly V1.23.0 and added the result in below sheet.

**Sheet Link:**
https://docs.google.com/spreadsheets/d/1LcYXmQcuLTBIHZrzUjoYaUvnzisq2ReRvxVwmn-xGEA/edit#gid=1030593634

**Bug Link;** [1641](https://github.com/valora-inc/wallet/issues/1641)

**Devices:** iPhone 13 mini (15.1.1), One plus 7t (11.0)
CC: @gnardini
Thanks..!!

ValoraQA commented:
Hey @MuckT, We have verified this task using Latest iOS Test flight build V1.24.1, Android Internal Build V1.24.1 and added the result in below sheet.

**Sheet Link:**
https://docs.google.com/spreadsheets/d/15nGo3iJAEVJ76wHae9PvYuWI7S2bovKAQPX_edtwPEI/edit#gid=0

**Bug Link:** [1641](https://github.com/valora-inc/wallet/issues/1641)

**Devices:** Google Pixel 2XL(11.0) , Google Pixel (10.0) , iPhone 12(14.7), iPhone 13 Mini (15.1.1)

CC: @gnardini
Thanks..!!

ValoraQA commented:
Hey @MuckT, We have verified this task using Latest iOS Test flight build V1.24.1 and added the result in below sheet.

**Sheet Link:**
https://docs.google.com/spreadsheets/d/15nGo3iJAEVJ76wHae9PvYuWI7S2bovKAQPX_edtwPEI/edit#gid=0

**Devices:** iPhone XS Max (13.5.1)
CC: @gnardini
Thanks..!!

ValoraQA commented:
Hey @MuckT, We have verified this task using Latest Android Internal release build V1.24.1 and added the result in below sheet.

**Sheet Link**:
https://docs.google.com/spreadsheets/d/15nGo3iJAEVJ76wHae9PvYuWI7S2bovKAQPX_edtwPEI/edit#gid=0

**Devices** : Samsung Galaxy A8+(8.0)
CC: @gnardini 
Thanks..!!

ValoraQA commented:
Hey @MuckT, We have verified this task using Latest Android Internal release build V1.24.1 and added the result in below sheet.

**Sheet Link**:
https://docs.google.com/spreadsheets/d/15nGo3iJAEVJ76wHae9PvYuWI7S2bovKAQPX_edtwPEI/edit#gid=0

**Bug Link:** [1641](https://github.com/valora-inc/wallet/issues/1641)

**Devices:** Redmi Note 5 Pro (9.0)

CC: @gnardini 
Thanks..!!
